### PR TITLE
Add Pytorch-UNet model

### PR DESCRIPTION
**Pull request by rajkar86**
Code repo for model: https://github.com/milesial/Pytorch-UNet.git
*Note: I noticed that the model has been changed since the [fork](https://github.com/supriyar/Pytorch-UNet) for the previous attempt to add this model. So I used the (original repository)[https://github.com/milesial/Pytorch-UNet.git].*

#### Test plan

Tested using run.py for all combinations of `{train,eval}` and `{cpu,cuda}` devices and `{jit,eager}` modes

```
Running train method from pytorch_unet on cpu in eager mode
Ran in 3.708993434906006 seconds.
Running train method from pytorch_unet on cuda in eager mode
Ran in 0.06084632873535156 seconds, gpu time 129.12742614746094.
Running train method from pytorch_unet on cpu in jit mode
Ran in 3.7224016189575195 seconds.
Running train method from pytorch_unet on cuda in jit mode
Ran in 0.1113424301147461 seconds, gpu time 180.25369262695312.
Running eval method from pytorch_unet on cpu in eager mode
Ran in 1.3530182838439941 seconds.
Running eval method from pytorch_unet on cuda in eager mode
Ran in 0.00830221176147461 seconds, gpu time 20.900863647460938.
Running eval method from pytorch_unet on cpu in jit mode
Ran in 1.4119977951049805 seconds.
Running eval method from pytorch_unet on cuda in jit mode
Ran in 0.06007719039916992 seconds, gpu time 73.89491271972656.
```

### CODE REVIEW

File: torchbenchmark/models/pytorch_unet/__init__.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        self.sample_masks = torch.randint(0, 1, (1, 640, 959), dtype=torch.int64).to(self.device)
+        self.model = UNet(n_channels=3, n_classes=2, bilinear=True).to(self.device)
+        if self.jit:
+            self.model = torch.jit.script(self.model)
+
+    def get_module(self):
+        return self.model, (self.sample_inputs,)
+
+    def train(self, niter=1):
+        optimizer = optim.RMSprop(self.model.parameters(), lr=self.args.lr, weight_decay=1e-8, momentum=0.9)
```
Comment (xuzhao9) [inline torchbenchmark/models/pytorch_unet/__init__.py:43]: Can we move these three lines to `__init__()`?

File: torchbenchmark/models/pytorch_unet/__init__.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    task = COMPUTER_VISION.SEGMENTATION
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+
+        self.args = self._get_args()
+        self.sample_inputs = torch.rand((1, 3, 640, 959), dtype=torch.float32).to(self.device)
```
Comment (xuzhao9) [inline torchbenchmark/models/pytorch_unet/__init__.py:33]: Is the shape of the sample_inputs here the same as the original data input?

File: torchbenchmark/models/pytorch_unet/pytorch_unet/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+*.pyc
```
Comment (xuzhao9) [inline torchbenchmark/models/pytorch_unet/pytorch_unet/.gitignore:1]: Please manually remove this file and the `.github` directory as we are managing gitignores globally.

**Review by xuzhao9 — APPROVED** [review]
LGTM overall. Please address the inline comments.

File: torchbenchmark/models/pytorch_unet/__init__.py (hunk 2 above)
Comment (rajkar86) [reply torchbenchmark/models/pytorch_unet/__init__.py:33]: Yes, I printed the shape by downloading the actual data referred in the original repo and running train.

File: torchbenchmark/models/pytorch_unet/pytorch_unet/.gitignore (hunk 3 above)
Comment (rajkar86) [reply torchbenchmark/models/pytorch_unet/pytorch_unet/.gitignore:1]: Will do.

File: torchbenchmark/models/pytorch_unet/__init__.py (hunk 2 above)
Comment (eircfb) [reply torchbenchmark/models/pytorch_unet/__init__.py:33]: Add a comment explaining where the shape came from, link to data, etc. Ideally you can describe what the sizes mean. For example is first size the batch_size?

Comment (rajkar86) [reply torchbenchmark/models/pytorch_unet/__init__.py:37]: Added comments.

**Pull request closed without merging**